File: java/src/main/java/org/nitk/hackathon/utils/UPIPayment.java
package org.nitk.hackathon.utils;

public class UPIPayment implements Payment {
    private String upiId;
    public UPIPayment(String upiId) {
        this.upiId = upiId;
    }

    @Override
    public void pay(double amount) {
        //logic
    }

    @Override
    public Payment copy() {
        return new UPIPayment(upiId);
    }

    public void setUpiId(String upiId) {
        this.upiId = upiId;
    }

}

File: python/utils/upi_payment.py

from python.utils.payment import Payment


class UPIPayment(Payment):
    def __init__(self, upi_id):
        self.upi_id = upi_id

    def pay(self, amount):
        # Logic for paying via UPI can go here
        print(f"Paid {amount} via UPI ({self.upi_id})")

    def copy(self):
        return UPIPayment(self.upi_id)

    def set_upi_id(self, upi_id):
        self.upi_id = upi_id
